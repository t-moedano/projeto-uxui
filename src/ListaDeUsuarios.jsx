import React, { useEffect, useState } from 'react';
import NumberFormat from 'react-number-format';
import './listadeusuarios.css';
import axios from 'axios';

//Pegando as informações da API pelo GET
const ListaDeUsuarios = () => {
    const [infos, setInfos] = useState([])
    useEffect(() => {
        axios.get('https://www.mocky.io/v2/5d531c4f2e0000620081ddce', {
            method: 'GET',
        }).then((resposta) => {setInfos(resposta.data)})
    }, [])

// Mock com lista de cartões para teste
const cards = [
    // cartão válido
    {
      card_number: '1111111111111111',
      cvv: 789,
      expiry_date: '01/18',
    },
    // cartão inválido
    {
      card_number: '4111111111111234',
      cvv: 123,
      expiry_date: '01/20',
    },
];

// Função para pegar a escolha do cartão do input select
const escolhaDoCartao = (event) => {
    setValorCartao(event.target.value);
}

// Ações dos modals
const [abrirPagamento, setAbrirPagamento] = useState("none"); // Para abrir modal de pagamento
const [pegarUsuario, setPegarUsuario] = useState(""); // Para pegar o nome do usuário
const [abrirPagou, setAbrirPagou] = useState("none"); // Para abrir modal com recibo de pagamento
const [abrirNaoRecebeu, setAbrirNaoRecebeu] = useState(""); // Para msg de erro de pagamento
const [valorCartao, setValorCartao] = useState("1"); // Para pegar o cartão escolhido para pagamento
const [valorDinheiro, setValorDinheiro] = useState(""); // Para pegar o valor de pagamento digitado
const [validarCampo, setValidarCampo] = useState("none"); // Para validar campo de valor digitado

// Função para abrir o modal de pagamento do usuário
const abrirModalPagar = (name) => {
    setAbrirPagamento("flex")
    setPegarUsuario(name)
}

// Função que abre o modal de recibo de pagamento 
const abrirModalPagou = () => {
    if (valorDinheiro === "") {
        setValidarCampo("flex");
    } else 
        {
        if (valorCartao === "1") {
            setAbrirNaoRecebeu("");
        } else {
            setAbrirNaoRecebeu("não");
        }
        setAbrirPagamento("none");
        setAbrirPagou("flex");
        setValorDinheiro("");
        setValidarCampo("none");
    }
}

// Função para fechar o modal do recibo de pagamento
const fecharModalPagamento = () => {
    setAbrirPagamento("none");
}

// Função para fechar o modal do recibo de pagamento
const fecharModal = () => {
    setAbrirPagou("none");
}

// Função para validar campo de valor para pagamento do usuário
const valorInput = (event) => {
    setValorDinheiro(event.target.value);
    setValidarCampo("none");
}

// Renderizando na tela as informações recebidas da API 
    return (
        <>
            <div className="clientGrid">
                {infos.map(item => (
                    <div className="clientcardcontainer" key={item.index}>
                        <div className="clientcard" onClick={()=>{abrirModalPagar(item.name)}}>
                            <img className="image" src={item.img} alt="Foto do usuário" />
                            <div className="client-info">
                                <div className="info">Nome do Usuário: {item.name}</div>
                                <div className="info">ID: {item.id} - Username: {item.username}</div>
                            </div>
                            <div className="hover">
                            </div>
                            <div className="pay-button">
                                <a className="pay-text">PAGAR</a>
                            </div>
                        </div>
                    </div>
                ))}
            </div>

            {/*--------------------------------Abrir Modal de pagamento----------------------------------*/}
            <div className="overlay" style={{display: abrirPagamento}}>       
                <div className="abrirModal">
                    <p className="texto-cabecalho-modal">Pagamento para <span>{pegarUsuario}</span></p>
                    <div className="gridchoice">
                        <div>Digite o valor:</div>
                        <div className="valorInput">
                            <NumberFormat thousandSeparator={true} value={valorDinheiro} onChange={valorInput} prefix={'R$ '} inputMode="numeric" placeholder="R$ 0,00"/>
                            <p style={{display:validarCampo}}>Campo obrigatório</p>
                        </div>
                        <div>
                            Escolha um cartão:
                        </div>
                        <select className="cardselector" value={valorCartao} onChange={escolhaDoCartao}>
                            <option value="1">Cartão com final {cards[0].card_number.substr(-4)}</option>
                            <option value="2">Cartão com final {cards[1].card_number.substr(-4)}</option>
                        </select>
                    </div>

                    <div className="gridbutton">
                        <div className="pay-button pay-text cancel" onClick={()=>{fecharModalPagamento()}}>CANCELAR</div>
                        <div className="pay-button pay-text success" onClick={()=>{abrirModalPagou ()}}>PAGAR</div>
                    </div>
                </div>  
            </div>
            {/*------------------------------Abrir Modal de recibo de pagamento--------------------------------*/}
            <div className="abrirModal" style={{display: abrirPagou}}>
                <p className="texto-cabecalho-modal">Recibo de pagamento</p>
                <p>O Pagamento <b>{abrirNaoRecebeu}</b> foi concluído com sucesso</p>
                <a className="pay-button pay-text" onClick={()=>{fecharModal()}}>Fechar</a>
            </div>
        </>
    )
}

export default ListaDeUsuarios