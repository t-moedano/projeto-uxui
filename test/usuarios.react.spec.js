import React from 'react';
import renderer from 'react-test-renderer';
import ListaDeUsuarios from '../src/ListaDeUsuarios';
import {cleanup, fireEvent, screen, render} from '@testing-library/react';

afterEach(cleanup);


it('Snapshot test for content', () => {
  const component = renderer.create(
    <ListaDeUsuarios />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


it('Test Open Pay Modal', () => {
  render(
    <ListaDeUsuarios />,
  );
  fireEvent.click(screen.getByText(/PAGAR/i))
  const text = screen.getByText(/Digite o valor:/i);
  expect(text).toBeTruthy();
})

