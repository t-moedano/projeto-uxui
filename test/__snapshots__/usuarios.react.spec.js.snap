// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for content 1`] = `
Array [
  <div
    className="overlay"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="abrirModal"
    >
      <p
        className="texto-cabecalho-modal"
      >
        Pagamento para 
        <span>
          
        </span>
      </p>
      <div
        className="gridchoice"
      >
        <div>
          Digite o valor:
        </div>
        <div
          className="valorInput"
        >
          <input
            inputMode="numeric"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            placeholder="R$ 0,00"
            type="text"
            value=""
          />
          <p
            style={
              Object {
                "display": "none",
              }
            }
          >
            Campo obrigatório
          </p>
        </div>
        <div>
          Escolha um cartão:
        </div>
        <select
          className="cardselector"
          onChange={[Function]}
          value="1"
        >
          <option
            value="1"
          >
            Cartão com final 
            1111
          </option>
          <option
            value="2"
          >
            Cartão com final 
            1234
          </option>
        </select>
      </div>
      <div
        className="gridbutton"
      >
        <div
          className="pay-button pay-text cancel"
          onClick={[Function]}
        >
          CANCELAR
        </div>
        <div
          className="pay-button pay-text success"
          onClick={[Function]}
        >
          PAGAR
        </div>
      </div>
    </div>
  </div>,
  <div
    className="abrirModal"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p
      className="texto-cabecalho-modal"
    >
      Recibo de pagamento
    </p>
    <p>
      O Pagamento 
      <b>
        
      </b>
       foi concluído com sucesso
    </p>
    <a
      className="pay-button pay-text"
      onClick={[Function]}
    >
      Fechar
    </a>
  </div>,
]
`;
